tests/test_api_render.py

```diff
@@ -182,6 +182,31 @@ def test_cross_info_index_platform(testing_config):
     assert metadata.config.host_platform == info_index["platform"]
 
 
+def test_noarch_with_platform_deps(testing_workdir, testing_config):
+    recipe_path = os.path.join(metadata_dir, "_noarch_with_platform_deps")
+    build_ids = {}
+    for subdir_ in ["linux-64", "linux-aarch64", "linux-ppc64le", "osx-64", "win-64"]:
+        platform, arch = subdir_.split("-")
+        m = api.render(
+            recipe_path, config=testing_config, platform=platform, arch=arch
+        )[0][0]
+        build_ids[subdir_] = m.build_id()
+
+    assert len(set(build_ids.values())) == 3
+    assert build_ids["linux-64"] == build_ids["linux-aarch64"]
+    assert build_ids["linux-64"] == build_ids["linux-ppc64le"]
```

Comment:
```suggestion
    # one hash for each platform, plus one for the archspec selector
    assert len(set(build_ids.values())) == 4
    assert build_ids["linux-64"] == build_ids["linux-aarch64"]
    assert build_ids["linux-64"] != build_ids["linux-ppc64le"]  # not the same due to archspec
```

@beeankha Can we commit [this suggestion](https://github.com/conda/conda-build/pull/4606/files#r1060469957) together with this one? Thanks 🙏